.github/workflows/build-with-clang.yml

```diff
@@ -52,7 +53,7 @@ jobs:
       - name: Install mkl_random dependencies
         shell: bash -l {0}
         run: |
-          pip install cython setuptools pytest pytest-cov
+          pip install cython setuptools">=70.1" wheel
```

Comment:
There is `setuptools">=77"` in other places. Was it intended to have a bit different dependency?

Reply:
`>=77` avoids a warning in builds, `>=70.1` is enough to make the builds work. It can be changed to be consistent.